app/uploaders/media_item_uploader.rb

```diff
@@ -0,0 +1,91 @@
+# encoding: utf-8
+
+class MediaItemUploader < CarrierWave::Uploader::Base
+  include CarrierWave::MiniMagick
+  include CarrierWave::MimeTypes
+
+  IMAGE_REGEX = /\Aimage\/.*\z/i
+  VIDEO_REGEX = /\Avideo\/.*\z/i
+
+  # the kind of storage to use for this uploader
+  storage :file
+
+  process :set_content_type
+  process :set_size_and_type_on_model
+  process :set_media_kind_on_model
+  process :set_height_and_width_on_model, if: :image?
+
+  version :thumb, if: :image? do
+    process resize_to_fill: [100, 100]
+  end
+  #TODO: Identify appropriate dimensions for these versions
+  version :small
+  version :medium
+  version :large
+
+  # the directory where uploaded files will be stored.
+  def store_dir
+    owner_type = model.media_attachable_type.underscore
+    owner_id = model.media_attachable_id
+    path = File.join('uploads', Rails.env, owner_type, "#{owner_id}", "#{model.id}")
```

Comment:
It was either that or an explicit cast to string, I decided the interpolation was cleaner.